subgraphs/erc20/src/mappings/token.ts

```diff
@@ -320,20 +320,15 @@ function handleTransferEvent(
     if (isNewAccount(destination)) {
       // It means the receiver is a new holder
       toAddressIsNewHolderNum = BIGINT_ONE;
-    } else {
-      balance = getOrCreateAccountBalance(
-        getOrCreateAccount(destination),
-        token
-      );
-      if (balance.amount == BIGINT_ONE) {
-        // It means the receiver's token balance is 0 before transferal.
-        toBalanceIsZeroNum = BIGINT_ONE;
-      }
+    }
+    balance = getOrCreateAccountBalance(getOrCreateAccount(destination),token);
+    if (balance.amount == BIGINT_ZERO) {
+      // It means the receiver's token balance is 0 before transferal.
+      toBalanceIsZeroNum = BIGINT_ONE;
     }
 
     token.currentHolderCount = token.currentHolderCount
       .minus(FromBalanceToZeroNum)
-      .plus(toAddressIsNewHolderNum)
```

Comment:
is this because `toBalanceIsZeroNum` will already count a new toAddress?

Reply:
yes, because the earlier if else was simplified. `toAddressIsNewHolderNum` is not used in this calculation anymore because it doesnt matter if the addresses is a new holder or not